### PR TITLE
Bugfix: NaN in the wfc_centers and spreads when compiling with gfortran 2017

### PR DESCRIPTION
**Pull request by nscolonna**
Fixed setting the reference point for the centers calculation to 0.0 Might be improved.

### CODE REVIEW

File: quantum_espresso/kcp/CPV/nksiclib.f90 [hunk 1]
```diff
@@ -7499,7 +7499,8 @@ SUBROUTINE compute_nksic_centers(nnrx, nx, nudx, nbsp, nspin, iupdwn, &
       !
       CALL ions_cofmass(taus, pmass, na, nsp, rs)
       ! and use it as reference position
-      !r0=0.d0
+      r0=0.d0 ! FIXME: NsC not sre why rhis was commented out. Need to check what we need here (15/09/2022)
+      !                For now this is fine to avoid NaN when compiling with gfortran 2017
```
Comment (elinscott) [inline quantum_espresso/kcp/CPV/nksiclib.f90:7503]: Do you know why we have two separate variables `r0` and `rs` and don't simply use `ions_cofmass` to initialise `r0`?

Comment (nscolonna) [reply quantum_espresso/kcp/CPV/nksiclib.f90:7503]: No idea, I agree with you we should use rs( i.e. the output of ions_cofmass. this should be the center of mass of the system. OK for molecule not sure for solids).  No idea why and when r0 was introduced. Now I just checked that setting r0=0 I can recover the results of the (bugged, but running) intel compiled version. 
This is why I uncommented it as a quick fix.
Also we never really use the centers and the R^2 in the workflow or the fortran calculation I guess.

Comment (elinscott) [reply quantum_espresso/kcp/CPV/nksiclib.f90:7503]: I can update the test suite if necessary

Comment (nscolonna) [reply quantum_espresso/kcp/CPV/nksiclib.f90]: Thanks! Just run the test suite on my laptop with the latest commit (using ion_cofmass to initialize r0) and it looks all good:
(koopmans) nicola@nicola-XPS-13-9360:~/Scrivania/CODES/git/koopmans$ make tests 
python3 -m pytest -m "not tutorials" tests/
========================================================================= test session starts =========================================================================
platform linux -- Python 3.7.10, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/nicola/Scrivania/CODES/git/koopmans, configfile: pytest.ini
plugins: hypothesis-6.54.3, cov-3.0.0
collected 63 items / 13 deselected / 50 selected                                                                                                                      

tests/test_cell.py ..                                                                                                                                           [  4%]
tests/test_commands.py .....                                                                                                                                    [ 14%]
tests/test_kpoints.py ..                                                                                                                                        [ 18%]
tests/test_pseudopotentials.py .                                                                                                                                [ 20%]
tests/test_qei_to_json.py ..                                                                                                                                    [ 24%]
tests/test_utils.py ....                                                                                                                                        [ 32%]
tests/calculators/test_koopmans_cp.py .....                                                                                                                     [ 42%]
tests/calculators/test_ph.py .                                                                                                                                  [ 44%]
tests/calculators/test_projwfc.py .                                                                                                                             [ 46%]
tests/calculators/test_ui_calc.py ..                                                                                                                            [ 50%]
tests/io/test_io_calculators.py ..........                                                                                                                      [ 70%]
tests/io/test_io_json.py .                                                                                                                                      [ 72%]
tests/workflows/test_convergence.py .                                                                                                                           [ 74%]
tests/workflows/test_dft.py ..                                                                                                                                  [ 78%]
tests/workflows/test_singlepoint.py ........                                                                                                                    [ 94%]
tests/workflows/test_ui.py .                                                                                                                                    [ 96%]
tests/workflows/test_wannierize.py ..                                                                                                                           [100%]

========================================================================== warnings summary ===========================================================================
../../../../venvs/koopmans/lib/python3.7/site-packages/ase/__init__.py:15
  /home/nicola/venvs/koopmans/lib/python3.7/site-packages/ase/__init__.py:15: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    if LooseVersion(np.__version__) < '1.9':

../../../../venvs/koopmans/lib/python3.7/site-packages/setuptools/_distutils/version.py:346
  /home/nicola/venvs/koopmans/lib/python3.7/site-packages/setuptools/_distutils/version.py:346: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    other = LooseVersion(other)

tests/workflows/test_singlepoint.py::test_singlepoint_si_ki_dfpt_explicit
  /home/nicola/Scrivania/CODES/git/koopmans/koopmans/utils/_warnings.py:43: UserWarning: Minor disagreement with benchmark detected for hamiltonian/kc
    ki_eigenvalues_on_grid = 10.05490 differs from benchmark 10.05480 by 1.00e-04
    warnings.warn(message, cls)

tests/workflows/test_singlepoint.py::test_singlepoint_si_ki_dfpt_explicit
  /home/nicola/Scrivania/CODES/git/koopmans/koopmans/utils/_warnings.py:43: UserWarning: Minor disagreement with benchmark detected for postproc/emp/ki
    band structure[16] = 7.77594 differs from benchmark 7.77940 by 3.46e-03
    warnings.warn(message, cls)

tests/workflows/test_singlepoint.py::test_singlepoint_gaas_wan2odd
  /home/nicola/Scrivania/CODES/git/koopmans/koopmans/utils/_warnings.py:43: UserWarning: Minor disagreement with benchmark detected for postproc/occ/ki
    dos[513] = 35.80815 differs from benchmark 35.78208 by 2.61e-02
    warnings.warn(message, cls)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
===================================================== 50 passed, 13 deselected, 5 warnings in 1742.72s (0:29:02) ======================================================

File: quantum_espresso/kcp/CPV/nksiclib.f90 [hunk 2]
```diff
@@ -7497,9 +7497,8 @@ SUBROUTINE compute_nksic_centers(nnrx, nx, nudx, nbsp, nspin, iupdwn, &
       !
       ! compute ionic center of mass
       !
-      CALL ions_cofmass(taus, pmass, na, nsp, rs)
+      CALL ions_cofmass(taus, pmass, na, nsp, r0)
```
Comment (elinscott) [inline quantum_espresso/kcp/CPV/nksiclib.f90:7500]: Please remove the above definition of the internal variable `rs`

**Review by elinscott — CHANGES_REQUESTED** [review]
Looks great, the tests pass on my machine too. Please just remove the `rs` and then we can merge.

**Review by elinscott — APPROVED** [review]
Looks great!

**Pull request merged**